L1Trigger/L1THGCal/interface/backend/HGCalMulticlusteringImpl.h

```diff
@@ -38,12 +41,38 @@ class HGCalMulticlusteringImpl{
                      const HGCalTriggerGeometryBase & triggerGeometry
                      );
 
+    void clusterizeHistoMax( const std::vector<edm::Ptr<l1t::HGCalCluster>> & clustersPtr,
+		     l1t::HGCalMulticlusterBxCollection & multiclusters,
+		     const HGCalTriggerGeometryBase & triggerGeometry
+		     );
+
+    void clusterizeHistoThreshold( const std::vector<edm::Ptr<l1t::HGCalCluster>> & clustersPtr,
+		     l1t::HGCalMulticlusterBxCollection & multiclusters,
+		     const HGCalTriggerGeometryBase & triggerGeometry
+		     );
+
+
 private:
 
     void findNeighbor( const std::vector<std::pair<unsigned int,double>>&  rankedList,
                        unsigned int searchInd,
                        const std::vector<edm::Ptr<l1t::HGCalCluster>> & clustersPtr, 
                        std::vector<unsigned int>& neigbors);
+
+    std::map<std::vector<int>,float> fillHistoClusters( const std::vector<edm::Ptr<l1t::HGCalCluster>> & clustersPtrs );
```

Comment:
Since the key has a fixed number of values (3), a `std::tuple` would be more appropriate than a vector. Also since this data type is used in multiple places it would be simple to use a `typedef`

Reply:
Though using `std::tuple` might be painful. Maybe a `std::array` would be more convenient.